casepro/profiles/__init__.py

```diff
@@ -66,11 +66,24 @@ def _user_update_role(user, org, role, partner=None):
     elif role not in PARTNER_ROLES and partner:
         raise ValueError("Cannot specify a partner for role %s" % role)
 
-    # remove user from any partners for this org
-    user.partners.remove(*user.partners.filter(org=org))
+    # store the user's previous partner
```

Comment:
Seems like this could all be replaced by something like...

```python
remove_from = [p for p in user.partners_primary.filter(org=org) if p != partner]
user.partners_primary.remove(*remove_from)
```